### PR TITLE
Fix #973, Fix #1225: Sync text updates.

### LINKED ISSUE #973: Rename Sync to Brave Sync on settings screen
This String needs to be translated so we won't include this in 1.9 release

54379775-6cf04080-4660-11e9-930a-473a1d016a9a

### LINKED ISSUE #1225: Implement sync warn message
Same as https://github.com/brave/brave-browser/issues/4132 and https://github.com/brave/browser-android-tabs/issues/1475

@anthonypkeane @jhreis this should be part of 1.10 as the text changes went with Android in the last release

### PR DESCRIPTION
**Pull request by iccub**
This fixes #973, fixes #1225

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

### Screenshots:

Zrzut ekranu 2019-07-10 o 17 11 19
Zrzut ekranu 2019-07-10 o 17 11 36

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable)

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]

**Comment by iccub** [comment]
Copy taken from https://github.com/brave/brave-core/pull/2259

File: BraveShared/BraveStrings.swift [hunk 1]
```diff
@@ -322,7 +322,7 @@ extension Strings {
...
     public static let SyncSwitchBackToCameraButton = NSLocalizedString("SyncSwitchBackToCameraButton", tableName: "BraveShared", bundle: Bundle.braveShared, value: "I'll use my camera...", comment: "Switch back to camera button")
-    public static let SyncAddDeviceWordsDescription = NSLocalizedString("SyncAddMobileWordsDescription", tableName: "BraveShared", bundle: Bundle.braveShared, value: "On your device, navigate to Brave Sync in the Settings panel and tap the button \"%@\". Enter the sync chain code words shown below.", comment: "Sync add device description")
+    public static let SyncAddDeviceWordsDescription = NSLocalizedString("SyncAddMobileWordsDescription", tableName: "BraveShared", bundle: Bundle.braveShared, value: "On your device, navigate to Brave Sync in the Settings panel and tap the button \"%@\". Enter the sync chain code words shown below. Treat this code like a password. If someone gets ahold of it, they can read and modify your synced data.", comment: "Sync add device description")
```
Comment (srirambv) [inline BraveShared/BraveStrings.swift]: This should be on  a new line

**Review by srirambv — CHANGES_REQUESTED** [review]

**Comment by iccub** [comment]
Updated

Zrzut ekranu 2019-07-10 o 17 19 46

**Review by srirambv — APPROVED** [review]
++

**Comment by iccub** [comment]
there's still some text needing an update, WIP

**Comment by iccub** [comment]
One more update I missed by using outdated spec document

Zrzut ekranu 2019-07-10 o 17 47 01

**Review by srirambv — APPROVED** [review]
++

**Pull request merged**